### PR TITLE
Misc: Compiler warnings fixes.

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

Misc: Compiler warnings fixes.

### Rationale behind Changes

Fixing compiler warnings.

### Suggested Testing Steps

CI passes, dump run on 456 no difference,  vulkan, hw renderers still work.

### CODE REVIEW

**Comment by F0bes** [comment]
You can drop the debugger commit. I've already fixed it.

**Comment by refractionpcsx2** [comment]
Not sure if the full screen ui function was supposed to be hooked up or not, probably should check with @stenzek

File: pcsx2/ImGui/ImGuiOverlays.cpp [hunk 1]
```diff
@@ -899,12 +899,12 @@ void SaveStateSelectorUI::Draw()
 		// Leave 2 lines for the legend
 		const float legend_margin = ImGui::GetFontSize() * 2.0f + ImGui::GetStyle().ItemSpacing.y * 3.0f;
 		const float padding = 10.0f * scale;
-		const s32 current_slot = s_current_slot.load(std::memory_order_acquire);
 
 		ImGui::BeginChild("##item_list", ImVec2(0, -legend_margin), false,
 			ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoInputs | ImGuiWindowFlags_NoTitleBar |
 				ImGuiWindowFlags_NoBackground);
 		{
+			const size_t current_slot = s_current_slot.load(std::memory_order_acquire);
```
Comment (stenzek) [inline pcsx2/ImGui/ImGuiOverlays.cpp:907]: This should be s32 (like above). i can be casted/truncated in the loop.

File: pcsx2/ImGui/FullscreenUI.cpp [hunk 2]
```diff
@@ -999,16 +998,6 @@ void FullscreenUI::DoReset()
 	});
 }
```
Comment (stenzek) [inline pcsx2/ImGui/FullscreenUI.cpp:1000]: DoChangeDisc() should be removed too, and the call site replaced with DoDiscChangeFromFile().

**Pull request merged**